src/chainstate/stacks/boot/pox.clar

```diff
@@ -380,6 +458,141 @@
+                             (delegator principal)
+                             (until-burn-ht (optional uint))
+                             (pox-addr (optional { version: (buff 1),
+                                                   hashbytes: (buff 20) })))
+    (begin
+      ;; must be called directly by the tx-sender or by an allowed contract-caller
+      (asserts! (check-caller-allowed)
+                (err ERR_STACKING_PERMISSION_DENIED))
+
+      ;; tx-sender principal must not be stacking
+      (asserts! (is-none (get-stacker-info tx-sender))
+        (err ERR_STACKING_ALREADY_STACKED))
+
+      ;; tx-sender must not be delegating
+      (asserts! (is-none (get-check-delegation tx-sender))
+        (err ERR_STACKING_ALREADY_DELEGATED))
+
+      ;; add delegation record
+      (map-set delegation-state
+        { stacker: tx-sender }
+        { amount-ustx: amount-ustx,
+          delegated-to: delegator,
+          until-burn-ht: until-burn-ht,
+          pox-addr: pox-addr })
+
+      (ok true)))
+
+;; Commit partially stacked STX.
+;;   This allows a stacker/delegator to lock fewer STX than the minimal threshold in multiple transactions,
+;;   so long as: 1. The pox-addr is the same.
+;;               2. This "commit" transaction is called _before_ the PoX anchor block.
+;;   This ensures that each entry in the reward set returned to the stacks-node is greater than the threshold,
+;;   but does not require it be all locked up within a single transaction
+(define-public (stack-aggregation-commit (pox-addr { version: (buff 1), hashbytes: (buff 20) })
+                                         (reward-cycle uint))
+  (let ((partial-stacked
+         ;; fetch the partial commitments
+         (unwrap! (map-get? partial-stacked-by-cycle { pox-addr: pox-addr, sender: tx-sender, reward-cycle: reward-cycle })
+                  (err ERR_STACKING_NO_SUCH_PRINCIPAL))))
+    ;; must be called directly by the tx-sender or by an allowed contract-caller
+    (asserts! (check-caller-allowed)
+              (err ERR_STACKING_PERMISSION_DENIED))
+    (let ((amount-ustx (get stacked-amount partial-stacked)))
+      (try! (can-stack-stx pox-addr amount-ustx reward-cycle u1))
+      ;; add the pox addr to the reward cycle
+      (add-pox-addr-to-ith-reward-cycle
+       u0
+       { pox-addr: pox-addr,
+         first-reward-cycle: reward-cycle,
+         num-cycles: u1,
+         amount-ustx: amount-ustx,
+         i: u0 })
+      ;; don't update the stacking-state map,
+      ;;  because it _already has_ this stacker's state
+      ;; don't lock the STX, because the STX is already locked
+      ;;
+      ;; clear the partial-stacked state
+      (map-delete partial-stacked-by-cycle { pox-addr: pox-addr, sender: tx-sender, reward-cycle: reward-cycle })
+      (ok true))))
+
+;; As a delegator, stack the given principal's STX using partial-stacked-by-cycle
+;; Once the delegator has stacked > minimum, the delegator should call stack-aggregation-commit
```

Comment:
I don't think "delegator" is the right word here -- the principal who is yielding control of the STX lockup to the delegate would be the "delegator," whereas the principal who carries out the STX lockup on behalf of one or more other principals is merely the "delegate."  This method is called by the delegate, not the delegator ;)

Reply:
Yep, that works for me -- updated.